x-pack/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -72,8 +103,16 @@ export class TaskRunnerFactory {
         );
 
         const requestHeaders: Record<string, string> = {};
+
         if (apiKey) {
-          requestHeaders.authorization = `ApiKey ${apiKey}`;
+          const subApiKeyResult = await getSubApiKey(spaceId, apiKey);
+          if (subApiKeyResult) {
+            const apiKeyHeaderValue = Buffer.from(
+              `${subApiKeyResult.id}:${subApiKeyResult.api_key}`
+            ).toString('base64');
+            requestHeaders.authorization = `ApiKey ${apiKeyHeaderValue}`;
+            subApiKeyId = subApiKeyResult.id;
+          }
```

Comment:
The creation of the sub API key should happen within `x-pack/plugins/actions/server/create_execute_function.ts`.